platform-sdk/docs/proposals/TSS-Ledger-Id/TSS-Ledger-Id.md

```diff
@@ -241,20 +269,159 @@ process for a node and the process of changing the consensus roster.
+Merkle Leaf by itself is appropriate.
+
+##### Consensus Rosters
+
+It is expected that the TSS-Roster proposal has introduced a 3rd roster in the state called the `Candidate Roster`
+in addition to the `Active Roster` and the `Previous Roster`. These rosters are stored in the state as their own
+singleton.
+
+This proposal extends the roster data format in two ways:
+
+1. Each `RosterEntry` has a field for the number of shares allocated to the node. This value will not get big and
+   can fit in a byte or short.
+2. Each `RosterEntry` has a new long term public EC key called a `tssEcKey` that is needed in the Groth21 algorithm to
+   key rosters.
+
+##### Key Material (TSS Data Map)
+
+The TSS Data Map is a singleton Merkle Leaf that is a combined map of all the key material for all rosters that are
+tracked and the votes by nodes that have validated the key material.
+
+Keys in the TSS Data Map have the following structure: (KeyType, RosterHash, SequenceNumber).
+
+1. KeyType - An enum that is one of TSS_MESSAGE or TSS_VOTE.
+2. RosterHash - The hash of the consensus roster that the message or vote is related to.
+3. SequenceNumber - The order in which the message or vote came to consensus.
+
+The value associated with a TSS_MESSAGE key is the pair (ShareId, TssMessage)
+
+1. ShareId - The id of the share that the message is related to.
+2. TssMessage - the raw bytes of the TssMessage
+
+The value associated with a TSS_VOTE key is a bit vector with the following interpretation
+
+* The order of bits from least significant bit to most significant bit corresponds to the numeric order of share ids
+  from least to greatest.
+* If a bit is set to 1, then the TssMessage for the corresponding ShareId was valid and contributed to a successful
+  reconstruction of the ledger id.
+* If a bit is set to 0, then the TssMessage for the corresponding ShareId was either invalid, was not received, or
+  was not used in the reconstruction of the ledger id.
+
+Lifecycle Invariants
+
+1. Inserts should only be made for candidate rosters that are going through the keying process.
+2. When a roster is no longer stored in the state, the corresponding key material should be removed.
+3. Every roster that has become active after the TSS Genesis process must have sufficient key material to be
+   able to recover the ledger id.
+
+#### New Wiring Components
+
+The following are new components in the system:
+
+1. TSS State Manager (Not Wired)
+2. TSS Message Creator (Wired)
+3. TSS Message Validator (Wired)
+4. TSS Key Manager (Wired)
+
+##### TSS State Manager
+
+The TSS state manager is executed on the transaction handling thread and is capable of reading and writing the state
+during round handling. The logic will be encapsulated in a class with a well-defined API, but this will not be a fully
+fledged wiring component outside the consensus round handler.
+
+Responsibilities:
```

Comment:
There seems to be a lot. While I understand that all of the below needs to happen, I think we might want to think about splitting the responsibilities across several components instead of having a single, very complex "TSS State Manager".

Reply:
It's not really a component in the sense of having its own internal state in memory and operating on its own thread scheduler.  This is more like a collection of methods that get invoked when handling the TSS related system transactions.